PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/controller_class/configuration_controller.cpp

```diff
@@ -0,0 +1,171 @@
+#include "configuration_controller.h"
+
+/**
+ * @brief Construct a new Configuration Controller and initializing some needed pointers to be used inside it 
+ */
+ConfigurationController::ConfigurationController()
+{
+    configuration = ConfigurationSpace::constructConfigurationSpace();
+    capability =  PCIECapability::constructPCIECapability();
+
+    handler = make_shared<ConfigurationRequestHandler>(configuration, capability);
+    tlpConstructor = make_shared<TLPConstructor>();
```

Comment:
why here you are constructing TLP, it is already implemented and you just use it?